--- a/target/linux/mediatek/filogic/base-files/etc/hotplug.d/ieee80211/11_fix_wifi_mac
+++ b/target/linux/mediatek/filogic/base-files/etc/hotplug.d/ieee80211/11_fix_wifi_mac
@@ -1,3 +1,8 @@
 #!/bin/sh
 
 case "$DEVPATH" in
+nradio,wt9103)
+	hw_mac_addr=$(mtd_get_mac_ascii bdinfo "fac_mac ")
+	[ "$PHYNBR" = "0" ] && echo "$hw_mac_addr" > /sys${DEVPATH}/macaddress
+	[ "$PHYNBR" = "1" ] && macaddr_setbit_la $hw_mac_addr > /sys${DEVPATH}/macaddress
+	;;
 esac